src/proto/h2/mod.rs

```diff
@@ -335,21 +339,34 @@ where
         cx: &mut Context<'_>,
         buf: &[u8],
     ) -> Poll<Result<usize, io::Error>> {
-        if let Poll::Ready(reset) = self.send_stream.poll_reset(cx) {
-            return Poll::Ready(Err(h2_to_io_error(match reset {
-                Ok(reason) => reason.into(),
-                Err(e) => e,
-            })));
-        }
         if buf.is_empty() {
             return Poll::Ready(Ok(0));
         }
         self.send_stream.reserve_capacity(buf.len());
-        Poll::Ready(match ready!(self.send_stream.poll_capacity(cx)) {
-            None => Ok(0),
-            Some(Ok(cnt)) => self.send_stream.write(&buf[..cnt], false).map(|()| cnt),
-            Some(Err(e)) => Err(h2_to_io_error(e)),
-        })
+
+        let cnt = match ready!(self.send_stream.poll_capacity(cx)) {
+            None => Some(0),
+            Some(Ok(cnt)) => self
+                .send_stream
+                .write(&buf[..cnt], false)
+                .ok()
```

Comment:
This will ignore the error, but that's on purpose, right?

Reply:
Yeah, that's the intent, as we will get the right error from `poll_reset` anyway.

Reply:
Ok got it. Perhaps a comment to that effect will help us when looking in this file a few months or years later 😅

Reply:
Done.